### PR TITLE
2.) Upload (new audio icon) #1395

### PR DESCRIPTION
**Pull request by Kamran-Khan-Gene**
image

image

image

image

image

image

https://github.com/Pylons-tech/pylons/issues/1395

## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my PR.

File: easel/lib/models/nft_format.dart [hunk 1]
```diff
@@ -26,13 +26,13 @@ class NftFormat {
   static List<NftFormat> get supportedFormats => [
         NftFormat(
           format: NFTTypes.image,
-          extensions: ['jpg', 'png', 'svg', 'heif', 'jpeg', 'gif'],
+          extensions: ['jpg', 'png', 'svg', 'heic'],
```
Comment (kjawadDeveloper2) [inline easel/lib/models/nft_format.dart:29]: Keep the other formats also here.

File: easel/lib/models/nft_format.dart [hunk 2]
```diff
@@ -44,8 +44,8 @@ class NftFormat {
         ),
         NftFormat(
           format: NFTTypes.audio,
-          extensions: ['wav', 'aiff', 'alac', 'flac', 'mp3', 'aac', 'wma', 'ogg'],
```
Comment (kjawadDeveloper2) [inline easel/lib/models/nft_format.dart:47]: Keep other formats also here.

File: easel/lib/screens/choose_format_screen.dart [hunk 3]
```diff
@@ -64,11 +64,29 @@ class _ChooseFormatScreenState extends State<ChooseFormatScreen> {
...
+    } else if (nftFormat.format == NFTTypes.audio) {
+      formatMustBe = LocaleKeys.audio_nft_must_be.tr();
+    } else if (nftFormat.format == NFTTypes.video) {
+      formatMustBe = LocaleKeys.video_nft_must_be.tr();
+    } else if (nftFormat.format == NFTTypes.threeD) {
+      formatMustBe = LocaleKeys.three_d_nft_must_be.tr();
+    } else if (nftFormat.format == NFTTypes.pdf) {
+      formatMustBe = LocaleKeys.pdf_nft_must_be.tr();
+    }
+    return '"$fileName" ${LocaleKeys.could_not_uploaded.tr()} $formatMustBe';
```
Comment (kjawadDeveloper2) [inline easel/lib/screens/choose_format_screen.dart:81]: Not good approach use argumented translations.

    return '"$fileName" ${LocaleKeys.could_not_uploaded.tr()} $formatMustBe';

File: easel/lib/screens/choose_format_screen.dart [hunk 4]
```diff
@@ -281,132 +317,80 @@ class _CardWidget extends StatelessWidget {
...
+                  style: TextStyle(
+                    color: Colors.white,
+                    fontSize: 12.sp,
+                    fontWeight: FontWeight.w800,
+                  ),
+                ),
+                SizedBox(height: 10.h),
+                Text(
+                  (nftTypes == NFTTypes.video || nftTypes == NFTTypes.audio)
+                      ? "• ${(kFileSizeLimitForAudiVideoInGB * 1000).toStringAsFixed(0)}MB limit"
```
Comment (kjawadDeveloper2) [inline easel/lib/screens/choose_format_screen.dart:366]: Audio instead of Audi.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Comment by Tanyaml** [comment]
@Amean @afti-githobo please review.

File: easel/lib/models/nft_format.dart [hunk 5]
```diff
@@ -27,12 +35,14 @@ class NftFormat {
         NftFormat(
           format: NFTTypes.image,
           extensions: ['jpg', 'png', 'svg', 'heif', 'jpeg', 'gif'],
+          extensionsUI: ['jpg', 'png', 'svg', 'heic',],
```
Comment (kjawadDeveloper2) [inline easel/lib/models/nft_format.dart:38]: Revert.

File: easel/lib/models/nft_format.dart [hunk 6]
```diff
@@ -27,12 +35,14 @@ class NftFormat {
...
           badge: SVGUtils.kSvgNftFormatImage,
           color: EaselAppTheme.kBlue,
         ),
         NftFormat(
           format: NFTTypes.video,
           extensions: ['mp4', 'mov', 'm4v', 'avi', 'hevc'],
+          extensionsUI: ['mp4'],
```
Comment (kjawadDeveloper2) [inline easel/lib/models/nft_format.dart:45]: Revert.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

File: easel/lib/models/nft_format.dart (hunk 5 above)
Comment (Kamran-Khan-Gene) [reply easel/lib/models/nft_format.dart:38]: In the updated figma some extensions are removed so I removed those extensions but you asked for adding them again so I created a new extension list so the UI matches the Figma.

image

Comment (heyitsjk) [reply easel/lib/models/nft_format.dart:38]: > In the updated figma some extensions are removed so I removed those extensions but you asked for adding them again so I created a new extension list so the UI matches the Figma.
> 
> 
> 
> image
> 
> 
> 
> 

Extension list should remain untouched from how it is now in-app

image

Comment (Kamran-Khan-Gene) [reply easel/lib/models/nft_format.dart:38]: ok I am reverting my changes regarding extension

**Review by kjawadDeveloper2 — APPROVED** [review]

**Comment by heyitsjk** [comment]
Don't remove extensions from Easel. The ones currently in-app are the correct ones needing to be displayed. 

Reference: 
image

**Review by faisalnaveedRNS — APPROVED** [review]

**Review by MikeSofaer — APPROVED** [review]

**Pull request merged**